tests/api_test.py

```diff
@@ -9629,5 +9630,19 @@ def test_autodidax_smoketest(self):
     autodidax_module = importlib.util.module_from_spec(spec)
     spec.loader.exec_module(autodidax_module)
 
+class GarbageCollectionTest(jtu.JaxTestCase):
+  def test_xla_gc_callback(self):
+    # https://github.com/google/jax/issues/14882
+    x_np = np.arange(10, dtype='int32')
+    x_jax = jax.device_put(x_np)
+    x_np_weakref = weakref.ref(x_np)
+
+    del x_np
+    del x_jax
+    gc.collect()
```

Comment:
Is gc.collect actually asynchronous?

If it is, you could use a gc callback to wait for its completion.

Reply:
OK, I'll try dropping the sleep

Reply:
Seems like it's working – maybe I assumed incorrectly about the asynchronousity